src/query/sql/src/planner/binder/bind_context.rs

```diff
@@ -321,6 +323,46 @@ impl BindContext {
         }
     }
 
+    pub fn search_column_position_recursively(
```

Comment:
maybe we should not do it recursively,  only match the current bind context otherwise throw errors.

Reply:
done

I am not sure, but it is no harm to start with the simple way.